### ISSUE TITLE
Support e2e in maya

### ISSUE THREAD

**Issue by AmitKumarDas** [openebs/openebs#1216]
Some of the goals of e2e support in maya are:

- Programmatic way to reproduce bugs
- Write logic to verify a feature for particular versions
  - Can be used to demo a feature well
- Should be easy to use by contributors
  - Can be used before sending a PR
- Should be easy to use by operators/users
  - To test a feature
  - To reproduce a bug
- Should be easy to modify
- Should be easy to build

To start off with, let the e2e try to reproduce the issue #1214 
More complicated cases can be reproducing the issue #1204

**Comment by AmitKumarDas** [comment]
### High Level Design Thoughts:
- Expose a CustomResource (CR) that defines a test scenario
- Build a controller w.r.t above CR & package it as a docker image
- Controller will be a wrapper over CLI based tools
- Below are some of the necessary tools packed in the docker image:
  - kubectl

### A test scenario schema definition i.e. K8s CustomResource
$ cat blr.yaml
```yaml
scenario:
  name: blr
  data:
    mapi:
      image:
    provisioner:
      image:
    sc:
      repcount:
      repimage:
  setup:
    - kubectl: blr-ops.yaml
    log:
      - header:
        exec:
        args:
          - path:
      - header:
        exec:
        args:
    verify:
      - name:
        exec:
        args:
          - path:
        eq:
        retry:
      - name:
        exec:
        args:
          - path:
        contains:
  run:
    - kubectl: blr-vol.yaml
    - kubectl: blr-app.yaml
    log:
      - header:
        exec:
        args:
          - path:
      - header:
        exec:
        args:
          - path:
    verify:
      - name:
        exec:
        args:
          - path:
        eq:
        retry:
      - name:
        exec:
        args:
          - path:
        contains:
        repeat:
  teardown:
    - kubectl: blr-vol.yaml
```

**Comment by AmitKumarDas** [comment]
Alternative spec which is a retrofit of https://github.com/aquasecurity/kube-bench:

```yaml
suite:
- id: 1.1
  text: "Check QOS"
  type: "qos"
  tests:
  - shell: <<kubectl or helm or any terminal command that is available in docker img>>
    vars: 
      - {{ go templating / helm templating }}
    checks:
      - id: 1.1.1
        text: "Ensure volume has minimum QOS settings"
        check: {{ go templating / helm templating }}
        remedy: "Volume should have a minimum CPU of X and minimum Memory of Y"
        status: <<success/fail determined after execution of this check>>
```

**Comment by AmitKumarDas** [comment]
People have started writing similar intent based test files already...
ref - https://github.com/lrills/helm-unittest
ref - https://medium.com/@PeiChenLiu/make-your-helm-charts-comfort-and-robust-with-unit-tests-2dd570e3174f

**Comment by AmitKumarDas** [comment]
This is currently  a WIP item & will be solved via this [issue#2343](https://github.com/openebs/openebs/issues/2343)
